mypy/semanal.py

```diff
@@ -5125,6 +5226,93 @@ def visit_match_stmt(self, s: MatchStmt) -> None:
                 guard.accept(self)
             self.visit_block(s.bodies[i])
 
+    def visit_type_alias_stmt(self, s: TypeAliasStmt) -> None:
+        self.statement = s
+        type_params: TypeVarLikeList = []
+        all_type_params_names = []
+        for p in s.type_args:
+            upper_bound = p.upper_bound or self.object_type()
+            fullname = self.qualified_name(p.name)
+            type_params.append(
+                (
+                    p.name,
+                    TypeVarExpr(
+                        p.name,
+                        fullname,
+                        [],
+                        upper_bound,
+                        default=AnyType(TypeOfAny.from_omitted_generics),
+                    ),
+                )
+            )
+            all_type_params_names.append(p.name)
```

Comment:
Doesn't this need to differentiate between all three TypeVar kinds as well? ParamSpec and TypeVarTuple are supported for `type` statements.

Reply:
Checking for `p.kind` and adding `ParamSpecExpr` seem to fix [Issue 1 + 2](https://github.com/python/mypy/pull/17233#issuecomment-2106031542).

Reply:
This is fixed now.